src/mon/FSCommands.cc

```diff
@@ -1353,6 +1353,17 @@ int FileSystemCommandHandler::is_op_allowed(
       return -ENOENT;
     }
 
+    for (auto& g : op->get_session()->caps.grants) {
+      if (not g.profile.empty()) {
+	for (auto& pg : g.profile_grants) {
+	  if ((pg.service == "fs" || pg.service == "mds") &&
+	       (pg.allow & MON_CAP_W)) {
+	    return 1;
+	  }
+	}
+      }
+    }
```

Comment:
Why not add this check to `MonCap::fs_name_capable`?

Reply:
Ramana too pointed that out. That's where it as originally located but I was a bit confused.  I've moved it back after reading some code around profiles and services caps.